### PR TITLE
Add sampler documentation

### PR DESCRIPTION
**Pull request by PhilipVinc**
This is based off #1202, so you should only look at the notebook and or generated docs

This PR adds a new section `the sampler module` to the documentation, discussing how to use samplers stand-alone (without the variational state) and discussing some of their settings.

### CODE REVIEW

**Comment by PhilipVinc** [comment]
The documentation can be previewed here: https://netket--1203.org.readthedocs.build/en/1203/docs/sampler.html

**Comment by PhilipVinc** [comment]
Guys unless you're against this I'd like to merge this.

Can surely be improved in the future, but would have prevented some questions from some students.

File: netket/sampler/base.py [hunk 1]
```diff
@@ -48,6 +48,14 @@ class Sampler(abc.ABC):
...
     Note that fields marked with `pytree_node=False` are treated as static arguments
     when jitting.
+
+    Subclasses should be netket dataclasses and they should define the `_init_state`,
+    `_reset` and `_sample_chain` methods which only accept positional arguments.
+    See those methods definition for their signature.
+
+    Notice that those methods are different from the API-entry point without the leading
+    underscore in order to allow us to share some pre-processing code between sampleers
+    and simplify the definition of a new sampler. It is also possible
```
Comment (femtobit) [inline netket/sampler/base.py:58]:
```suggestion
    Subclasses should be NetKet dataclasses and they should define the `_init_state`,
    `_reset` and `_sample_chain` methods which only accept positional arguments.
    See the respective method's definition for its signature.

    Notice that those methods are different from the API-entry point without the leading
    underscore in order to allow us to share some pre-processing code between samplers
    and simplify the definition of a new sampler.
```

File: docs/docs/sampler.ipynb [hunk 2]
```diff
@@ -514,14 +514,6 @@
...
     "samples, sampler_state = sampler.sample(log_pdf, pars, state=sampler_state, chain_length=100)"
    ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "410531e1-3f36-4b22-ab3c-321b71b3532a",
-   "metadata": {},
-   "outputs": [],
-   "source": []
```
Comment (femtobit) [inline docs/docs/sampler.ipynb:524]: This just removes an empty cell, right? Does `samplers.ipynb` need to be changed in this PR at all?

**Review by femtobit — CHANGES_REQUESTED** [review]

File: docs/docs/sampler.ipynb (hunk 2 above)
Comment (PhilipVinc) [reply docs/docs/sampler.ipynb:524]: I mistakenly merged `samplers.ipynb` in the previous PR, but it was not yet linked in the docs.
I had a few changes to samplers.ipynb, but most was already in the previous PR.

**Review by gcarleo — APPROVED** [review]

**Review by femtobit — APPROVED** [review]

**Pull request merged**